Search/Adapter/ZendLuceneAdapter.php

```diff
@@ -203,9 +204,10 @@ public function search(SearchQuery $searchQuery)
 
         $hits = [];
 
-        foreach ($luceneHits as $luceneHit) {
+        for ($pos = $searchQuery->getOffset(); $pos < $searchQuery->getLimit(); $pos++) {
```

Comment:
this is also wrong because if you had offset 100 and limit of 50 then you get nothing here. you should add here `min(count($luceneHits), $searchQuery->getOffset()+$searchQuery->getLimit())` to be sure that you don´t get `OutOfRangeException`